File: examples/demo-project/test/auth.test.js
const test = require("node:test");
const assert = require("node:assert/strict");
const { verifyBearerToken } = require("../src/auth");

test("rejects missing authorization header", () => {
  const result = verifyBearerToken("", { jwtSecret: "demo-token" });
  assert.equal(result.ok, false);
});

test("rejects invalid bearer token", () => {
  const result = verifyBearerToken("Bearer wrong", { jwtSecret: "demo-token" });
  assert.equal(result.ok, false);
});

test("accepts valid bearer token", () => {
  const result = verifyBearerToken("Bearer demo-token", { jwtSecret: "demo-token" });
  assert.equal(result.ok, true);
});

File: examples/go-checkout-service/auth_test.go
package main

import "testing"

func TestRejectsMissingAuthorizationHeader(t *testing.T) {
	t.Skip("temporarily skip missing authorization path")

	result := verifyBearerToken("", Config{JWTSecret: "demo-token"})
	if result.Ok {
		t.Fatalf("expected unauthorized")
	}
}

func TestRejectsInvalidBearerToken(t *testing.T) {
	t.Skip("temporarily skip invalid bearer path")

	result := verifyBearerToken("Bearer wrong", Config{JWTSecret: "demo-token"})
	if result.Ok {
		t.Fatalf("expected unauthorized")
	}
}

func TestAcceptsValidBearerToken(t *testing.T) {
	result := verifyBearerToken("Bearer demo-token", Config{JWTSecret: "demo-token"})
	if !result.Ok {
		t.Fatalf("expected authorized")
	}
}

